File: cocsim/buildings/colliders/collider.py
class Collider:
    def get_attack_area(self, attack_range: float) -> "Collider":
        """Returns attack area from which units with given attack range
        can attack this building.
        """
        ...

    def get_nearest_point(self, x: float, y: float) -> tuple[float, float]:
        """Returns point of collider nearest to (x, y)."""

    def __contains__(self, item: tuple[int, int]) -> bool:
        """Checks if item (x, y) is inside of collider."""
        ...

File: backend/cocsim/src/buildings/colliders/collider.rs
use nalgebra::Vector2;

pub trait Collider {
    /// Returns attack area `Collider` from which units with given
    /// `attack_range` can attack this building.
    fn attack_area(&self, attack_range: f32) -> Box<dyn Collider>;

    /// Returns point of collider nearest to `point`.
    fn nearest_point(&self, point: Vector2<f32>) -> Vector2<f32>;

    /// Checks if `point` is inside of collider.
    fn contains(&self, point: Vector2<f32>) -> bool;
}
